### PR TITLE
Time option disabled by default on the SimpleDateTime Custom FormIO Component

### PR DESCRIPTION
**Pull request by MichaelChung123**
# Description
Making the Time option disabled by default on the SimpleDateTime Custom FormIO Component. Added a checkbox field to enable it manually.

## Types of changes
Allows a better user experience by not letting the Time option be enabled by default.

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-183

**Comment by TimCsaky** [comment]
This works. i was wondering if we want a 'Enable Date Input' checkbox as well. so they can use this simple component for either date or time (or both) that's the way the advanced version works.

**Review by TimCsaky — APPROVED** [review]

**Review by loneil — CHANGES_REQUESTED** [review]
Don't merge, seem to be losing time, can investigate more
image

File: components/src/components/SimpleDateTime/editForm/Component.edit.time.ts [hunk 1]
```diff
@@ -1,4 +1,12 @@
 export default [
+    {
+        type: 'checkbox',
+        input: true,
```
Comment (TimCsaky) [inline components/src/components/SimpleDateTime/editForm/Component.edit.time.ts:4]: maybe this can be false and you dont need the other file change?

File: components/src/components/SimpleDateTime/Component.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 /* tslint:disable */
+import _ from 'lodash';
```
Comment (jujaga) [inline components/src/components/SimpleDateTime/Component.ts:2]: Is this a necessary import? I don't see this being used in any of the changed lines.

Comment (MichaelChung123) [reply components/src/components/SimpleDateTime/Component.ts]: Forgot to remove from tests. Added new commit removing it.

File: app/frontend/component-update.js [hunk 3]
```diff
@@ -23,17 +23,19 @@ try {
...
+      //   console.log(`${TITLE} not found under "${FORMIO_DIR}"`); // eslint-disable-line no-console
+      //   buildComponents();
+      //   deployComponents();
+      // } else if (fs.statSync(FORMIO_DIR).mtime < fs.statSync(COMPONENTS_DIR).mtime) {
+      //   console.log(`${TITLE} "${COMPONENTS_DIR}" directory has been modified`); // eslint-disable-line no-console
+      //   buildComponents();
+      //   deployComponents();
+      // } else {
+      //   console.log(`${TITLE} is present and up to date`); // eslint-disable-line no-console
+      // }
```
[the remark this answers is not in the document]
Comment (jujaga) [reply app/frontend/component-update.js:38]: Good catch. This shouldn't be commented out at all.

File: components/src/components/SimpleDateTime/Component.ts [hunk 4]
```diff
@@ -59,4 +59,12 @@ export default class Component extends (ParentComponent as any) {
...
         };
     }
+
+    constructor(component) {
+      super(component);
+      // if enableTime is set to false, manually add time fields since it get removed later on through Form IO
+      if (!this.component.enableTime) {
+        this.component.format = this.component.format.concat(' hh:mm a');
+      }
+    }
```
Comment (loneil) [inline components/src/components/SimpleDateTime/Component.ts:69]: Looks like this is causing an exception now? If I drag the component on the setttings bar on hover doesn't come up so it can't be edited. When moving it onto the form it throws this in the console

image

It's wierd, one time I tried it it referred to 'component' instead of 'tabs' but not sure where that's coming from

Not happening on any other versions of the app

File: components/src/components/SimpleContent/Component.ts [hunk 5]
```diff
@@ -31,4 +31,17 @@ export default class Component extends (ParentComponent as any) {
...
+      if(html.includes('<a href=')) {
+        result = html.replace(/<a href=/g, '<a target="_blank" href=');
+      }
+      return result;
+    };
+
+    constructor(component, options, data) {
+      super(component, options, data);
+      component.html = this.addTargetBlankToLinks(component.html);
+    }
```
Comment (loneil) [inline components/src/components/SimpleContent/Component.ts:46]: Seems to have broken the Static Text control? Does it work on your local. Anything I enter isn't saved any more on this control on the PR.

**Review by loneil — CHANGES_REQUESTED** [review]
Unless it's a wierd build issue the 2 components might have been broken by overloading constructors?

File: app/frontend/component-update.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (MichaelChung123) [reply app/frontend/component-update.js]: Mistake from commit. It has been reverted now.

File: components/src/components/SimpleDateTime/Component.ts (hunk 4 above)
Comment (MichaelChung123) [reply components/src/components/SimpleDateTime/Component.ts]: Fixed the constructor by keeping the original arguments for it. Error not showing up on local anymore when using the Simple Date/Time picker.

File: components/src/components/SimpleContent/Component.ts (hunk 5 above)
Comment (MichaelChung123) [reply components/src/components/SimpleContent/Component.ts]: Unintentional add from my commit. This has been reverted now.

File: components/src/components/SimpleDateTime/Component.ts [hunk 6]
```diff
@@ -1,62 +1,74 @@
 /* tslint:disable */
-import { Components } from 'formiojs';
+import _ from "lodash";
```
Comment (jujaga) [inline components/src/components/SimpleDateTime/Component.ts:2]: This got reintroduced again somehow.

File: components/src/components/SimpleDateTime/Component.ts [hunk 7]
```diff
@@ -1,62 +1,74 @@
...
+import { Components } from "formiojs";
 const ParentComponent = (Components as any).components.datetime;
-import editForm from './Component.form';
+import editForm from "./Component.form";
```
Comment (jujaga) [inline components/src/components/SimpleDateTime/Component.ts:5]: Not sure why linting didn't pick up on this, but strings are supposed to be single quoted, not double quoted. Looks like the latest refactor changed everything to double quotes for some reason.

**Review by jujaga — CHANGES_REQUESTED** [review]

File: components/src/components/SimpleDateTime/Component.ts (hunk 6 above)
Comment (MichaelChung123) [reply components/src/components/SimpleDateTime/Component.ts]: Fixed all Linting errors in next commit.

File: components/src/components/SimpleDateTime/Component.ts (hunk 7 above)
Comment (MichaelChung123) [reply components/src/components/SimpleDateTime/Component.ts]: Fixed all Linting errors in next commit.

Comment (MichaelChung123) [reply components/src/components/SimpleDateTime/Component.ts]: I believe that my formatter for VSCode was introducing these Linting errors. It switched to Prettier for some reason, so I switched it back to the original VSCode formatter.

**Review by jujaga — APPROVED** [review]

File: components/src/components/Common/Simple.edit.data.ts [hunk 8]
```diff
@@ -28,9 +28,9 @@ export default [
...
         data: {
             custom(context) {
-                var values = [];
+                const values = [];
                 values.push({ label: 'Any Change', value: 'data' });
-                context.utils.eachComponent(context.instance.options.editForm.components, function(component, path) {
+                context.utils.eachComponent(context.instance.options.editForm.components, (component, path) => {
                     if (component.key !== context.data.key) {
                         values.push({
                             label: component.label || component.key,
```
Comment (loneil) [inline components/src/components/Common/Simple.edit.data.ts:36]: Is this related to the date time control? Not sure what this change is doing

Comment (MichaelChung123) [reply components/src/components/Common/Simple.edit.data.ts:36]: I ran the Linter and this was just a Linting issue that needed to be resolved.

**Review by loneil — APPROVED** [review]
Looks good, datetime stuff seems to be working all fine now
